pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -608,6 +607,22 @@ func isNotFoundError(err error) bool {
 	return false
 }
 
+func isProjectNotFoundError(err error, projectID string) bool {
+	switch t := err.(type) {
+	case *googleapi.Error:
+		return t.Code == 404 && strings.Contains(t.Message, fmt.Sprintf("The resource 'projects/%s' was not found", projectID))
```

Comment:
is there any documentation around this error message, it makes me a little nervous because what if google changes the error body?

Reply:
I think https://google.aip.dev/193#:~:text=Otherwise%2C%20the%20content%20of%20Status.message%20must%20be%20stable%2C%20providing%20the%20same%20text%20with%20the%20same%20request%2Dspecific%20information. applies here as there is no ErrorInfo in the response.
```
{
  "error": {
    "code": 404,
    "message": "The resource 'projects/openshift-gce-develd' was not found",
    "errors": [
      {
        "message": "The resource 'projects/openshift-gce-develd' was not found",
        "domain": "global",
        "reason": "notFound"
      }
    ],
    "status": "NOT_FOUND"
  }
}
```

I have updated the function to look for 'projects/%s' and notFound reason. If google changes this I am eating my Fedora.